types/jsreport-core/index.d.ts

```diff
@@ -65,50 +84,192 @@ declare namespace JsReport {
         directory: string;
     }
 
-    interface Reporter {
-        use(extension: Extension | ExtensionDefinition): Reporter;
-    }
-
-    interface Reporter {
-        afterRenderListeners: ListenerCollection;
-        afterTemplatingEnginesExecutedListeners: ListenerCollection;
-        beforeRenderListeners: ListenerCollection;
-        documentStore: DocumentStore;
-        initializeListeners: ListenerCollection;
+    interface ReporterInstance {
+        defaults?: any;
+        options?: Configuration;
+        afterRenderListeners?: ListenerCollection;
+        afterTemplatingEnginesExecutedListeners?: ListenerCollection;
+        beforeRenderListeners?: ListenerCollection;
+        closeListeners?: ListenerCollection;
+        documentStore?: DocumentStore;
+        blobStorage?: BlobStorage;
+        initializeListeners?: ListenerCollection;
+        renderErrorListeners?: ListenerCollection;
         // it would be nice to add winston.LoggerInstance here
         // however adding import winston = require('winston') breaks exported enums
         logger: any;
```

Comment:
I noticed you keep `logger` required here.
And I could really can not find any advantage of make all properties optional here.

Reply:
good call